### PR TITLE
Engagement refactoring w awarding

### PR DESCRIPTION
**Pull request by eri-trabiccolo**
continue working on the engagement awarding

### CODE REVIEW

**Comment by thomasplevy** [comment]
Some notes:

1.  When viewing an earned cert as an admin I'd expect to see the "Edit Certificate" link/button in the WP Admin bar (and I don't)
1. When editing an earned cert / achievement (on the admin panel) the "Engagements" menu in the sidebar collapses. Ideally this would stay expanded
1. When awarding a cert/achiemvent (creating a new my_* manually on the admin panel) the post type language switches from "Award Certificate" to "Add new certificate". I think it should say "Award New Certificate" and "Edit Awarded Certificate"
1. When editing an awarded cert / engagement we should add the student's name/email and a link to their reporting screen in a metabox somewhere

**Comment by eri-trabiccolo** [comment]
I agree on everything -
1,2,4 were on my list too.
3 added. 
❤️

**Comment by eri-trabiccolo** [comment]
I'm actually also planning to rework the submit box and add our own JavaScript because the wp JS forces the wording (e.g. Publish, while I want Award).

**Comment by eri-trabiccolo** [comment]
> 1. When viewing an earned cert as an admin I'd expect to see the "Edit Certificate" link/button in the WP Admin bar (and I don't)
> 2. When editing an earned cert / achievement (on the admin panel) the "Engagements" menu in the sidebar collapses. Ideally this would stay expanded

Fixed
> 3. When awarding a cert/achiemvent (creating a new my_* manually on the admin panel) the post type language switches from "Award Certificate" to "Add new certificate". I think it should say "Award New Certificate" and "Edit Awarded Certificate"

Fixed

> 4. When editing an awarded cert / engagement we should add the student's name/email and a link to their reporting screen in a metabox somewhere

Fixed

Added it in the same metabox, not exactly happy of the look and feel to be honest, what do you think? Do you think it deserves its only metabox? Also consider that when your work will be done, there will be no other fields in the metabox, as both title (and content, for the achievement) and background will be the WordPress default one.

File: includes/class.llms.post-types.php [hunk 1]
```diff
@@ -1456,34 +1444,62 @@ public static function register_taxonomies() {
...
+	 *
+	 * By default the post types submenu items link to the post type list table,
+	 * we want them to point to the "Add New" screen.
 	 *
 	 * @since [version]
 	 *
 	 * @return void
 	 */
-	public static function add_earned_egnagements_submenu_links() {
+	public static function modify_earned_engagements_submenu_links() {
```
Comment (thomasplevy) [inline includes/class.llms.post-types.php]: I think this is better suited in this file: https://github.com/gocodebox/lifterlms/blob/trunk/includes/admin/class.llms.admin.menus.php

**Comment by eri-trabiccolo** [comment]
wanna merge this in your branch?

**Pull request merged**